### PR TITLE
487 Solidity Support Modifiers

### LINKED ISSUE #487: [Solidity Support] Modifiers
### Problem

The Hedera network supports solidity as one of the smart contracts development languages.
Occasionally items have shown up that surfaced in-equivalence at the language layer.

### Solution

Add examples for modifier operations

- Add `/contracts/solidity/modifiers/Modifiers.sol`
- Cover the tasklist below

```[tasklist]
### Modifier Tasks
- [ ] pure
- [ ] view
- [ ] payable
- [ ] constant
- [ ] immutable
- [ ] indexed
- [ ] anonymous
- [ ] virtual
- [ ] override
```

Ref https://docs.soliditylang.org/en/latest/cheatsheet.html#modifiers

### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by ebadiere**
Solidity modifiers coverage

Fixes #487 

- [ ] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

**Review by Nana-EC — COMMENTED** [review]
Fix license headers, others LG

File: contracts/solidity/modifiers/A.sol [hunk 1]
```diff
@@ -0,0 +1,9 @@
+// SPDX-License-Identifier: MIT
```
[the remark this answers is not in the document]
Comment (ebadiere) [reply contracts/solidity/modifiers/A.sol]: Updated.

File: test/solidity/modifiers/Modifiers.js [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+    it("Should return the message in the from the derived contract that overrides the virtual function", async function() {
+        expect(await derivedContract.getData()).to.equal(55);
+        expect(await derivedContract.show()).to.equal("This is the derived contract");
+    });
+    
+    it("Should return the message in the from ContractB that overrides the show function", async function() {
+        expect(await contractB.getData()).to.equal(79);
+        expect(await contractB.show()).to.equal("This is the overriding contract B");
+    });    
+
```
Comment (Nana-EC) [inline test/solidity/modifiers/Modifiers.js:138]: nit: a few of these have an extra line before the final closing bracket

**Review by Nana-EC — APPROVED** [review]
LG
Please update PR with details and correct assignments also

File: test/solidity/modifiers/Modifiers.js (hunk 2 above)
Comment (ebadiere) [reply test/solidity/modifiers/Modifiers.js:138]: Removed.

**Pull request merged**